pages/admin.js

```diff
@@ -0,0 +1,116 @@
+import Head from "next/head";
+import db from "../data/database";
+import { products } from "../data/products";
+import { ListProductComponent } from "../components/product/list";
+import { useState, useEffect } from "react";
+
+let clearPage = () => {
+  let form = document.getElementById("submitForm");
+  form.reset();
+};
+export default function Admin() {
+  const [productName, setName] = useState("");
+  const [productPrice, setPrice] = useState("");
+  const [productDescription, setDescription] = useState("");
+
+  const submitNewProduct = (e) => {
```

Comment:
wrapping these in useCallback is always a good idea because otherwise, react will constantly recreate this definition whenever something in the state changes.
Also, this type of helper function should be declared outside of the component. the more logic you can take out of the component and isolate elsewhere, the cleaner and easier to understand your components will be.